authorization-server/test/grant_type_code.js

```diff
@@ -1,164 +1,116 @@
-                  validate.validateAccessToken(response, body);
-                });
-                //Try to get the token again but we shouldn't be able to reuse the same code
-                helper.postOAuthCode(code,
-                  function (error, response, body) {
-                    validate.validateInvalidCodeError(response, body);
-                    done();
-                  }
-                );
-              }
-            );
-          }
-        );
-      }
-    );
-  });
-  it('should work with the authorization_code not asking for a refresh token', function (done) {
-    //Log into the OAuth2 server as bob
-    helper.login(
-      function (error, response, body) {
-        //Get the OAuth2 authorization code
-        helper.getAuthorization({},
-          function (error, response, body) {
-            //Assert that we have the ?code in our URL
-            assert.equal(response.req.path.indexOf("/?code="), 0);
-            var code = response.req.path.slice(7, response.req.path.length);
-            validate.validateAuthorizationCode(code);
-            //Get the token
-            helper.postOAuthCode(code,
-              function (error, response, body) {
-                validate.validateAccessToken(response, body);
-                //Get the user info
-                helper.getUserInfo(JSON.parse(body).access_token,
-                  function (error, response, body) {
-                    validate.validateUserJson(response, body);
-                  }
-                );
-                done();
-              }
-            );
-          }
-        );
-      }
-    );
-  });
-  it('should give an error with an invalid client id', function (done) {
-    helper.login(
-      function (error, response, body) {
-        //Get the OAuth2 authorization code
-        helper.getAuthorization({clientId: 'someinvalidclientid'},
-          function (error, response, body) {
-            //assert that we are getting an error code of 400
-            assert.equal(response.statusCode, 403);
-            done();
-          }
-        );
-      }
-    );
+
+  it('should work with the authorization_code not asking for a refresh token', (done) => {
+    // Log into the OAuth2 server as bob
+    helper.login(() => {
+      // Get the OAuth2 authorization code
+      helper.getAuthorization({}, (error, response) => {
+        // Assert that we have the ?code in our URL
+        assert.equal(response.req.path.indexOf('/?code='), 0);
+        const code = response.req.path.slice(7, response.req.path.length);
+        validate.validateAuthorizationCode(code);
+        // Get the token
+        helper.postOAuthCode(code, (codeError, codeResponse, codeBody) => {
+          validate.validateAccessToken(codeResponse, codeBody);
+          // Get the user info
+          helper.getUserInfo(JSON.parse(codeBody).access_token, (uErr, uRes, uBody) => {
+            validate.validateUserJson(uRes, uBody);
+          });
+          done();
```

Comment:
That `done()` is in the wrong spot.  It should be moved up to below the `validateUserJson`